service/telemetry.go

```diff
@@ -109,8 +110,22 @@ func (tel *colTelemetry) initOnce(col *Collector) error {
 
 	logger.Info("Setting up own telemetry...")
 
-	instanceUUID, _ := uuid.NewRandom()
-	instanceID := instanceUUID.String()
+	if resource == nil {
+		resource = map[string]string{}
+	}
+
+	var instanceID string
+	instanceID = resource[semconv.AttributeServiceInstanceID]
+	if instanceID == "" {
```

Comment:
Do we want to enforce this? I always felt that we should not have this by default.

Reply:
I think we need the default. Otherwise there is no way to differentiate metrics coming from different instances.